docs/security.rst

```diff
@@ -310,3 +310,20 @@ standard port 443, you'll need to configure that too. Be aware that super user p
     # Optionally, set the server to listen on the standard SSL port.
     web_server_port = 443
     base_url = http://<hostname or IP>:443
+
+Impersonation
+'''''''''''''
+
+Airflow has the ability to impersonate a unix user while running task
+instances based on the task's ``run_as_user`` parameter, which takes a user's name.
+
+*NOTE* For impersonations to work, Airflow must be run with `sudo` as subtasks are run
```

Comment:
Agreed, for v1 lets keep security out of impersonation (i.e. maintain the existing assumption that users can be trusted). For v2 lets do as you suggest, along with other proper security auditing/hardening.